pkg/runtime.v2/framework/plugins/jobset/builder.go

```diff
@@ -49,21 +50,55 @@ func NewBuilder(objectKey client.ObjectKey, jobSetTemplateSpec kubeflowv2.JobSet
 	}
 }
 
-func (b *Builder) ContainerImage(image *string) *Builder {
-	if image == nil || *image == "" {
-		return b
-	}
+// Trainer updates JobSet values for the trainer Job.
+func (b *Builder) Trainer(info *runtime.Info, trainJob *kubeflowv2.TrainJob) *Builder {
```

Comment:
This should be more composaple API similar to 

- https://github.com/kubeflow/training-operator/blob/3f7ec16ed701d1511e13b904d4f1cd927d7d1074/pkg/util.v2/testing/wrapper.go#L285 ->
- https://github.com/kubeflow/training-operator/blob/3f7ec16ed701d1511e13b904d4f1cd927d7d1074/pkg/util.v2/testing/wrapper.go#L342 ->
- https://github.com/kubeflow/training-operator/blob/3f7ec16ed701d1511e13b904d4f1cd927d7d1074/pkg/util.v2/testing/wrapper.go#L444 ->
- https://github.com/kubeflow/training-operator/blob/3f7ec16ed701d1511e13b904d4f1cd927d7d1074/pkg/util.v2/testing/wrapper.go#L454

Basically, one function corresponds to one fields.

Reply:
@tenzen-y Why do you want to de-compose every container parameter to a separate function ?
Would it be easier to logically split it by sections ? E.g. JobSet consists of Trainer, Model Initiailizer, Dataset Initiailizer.

Reply:
As we discussed with @tenzen-y offline, this will be implemented as future enhancement after we support Trainer, Dataset, and Model initializer builder in JobSet plugin.